src/basic/mount-util.c

```diff
@@ -689,3 +689,35 @@ int umount_verbose(const char *what) {
                 return log_error_errno(errno, "Failed to unmount %s: %m", what);
         return 0;
 }
+
+const char *mount_propagation_flags_to_string(unsigned long flags) {
+
+        switch (flags & (MS_SHARED|MS_SLAVE|MS_PRIVATE)) {
+
+        case MS_SHARED:
+                return "shared";
+
+        case MS_SLAVE:
+                return "slave";
+
+        case MS_PRIVATE:
+                return "private";
+        }
+
+        return NULL;
+}
+
+unsigned long mount_propagation_flags_from_string(const char *name) {
+
+        if (isempty(name))
```

Comment:
0 is also a valid value for flags, but we return 0 on error. I think this might lead to confusion down the road. All callers below special case the empty string, which leads to quite verbose code. `mount_propagation_flags_from_string` is never called with NULL or empty argument, except in test code. So this could be simplified. If this returned the parsed value in an output parameter, the return value could hold a negative error, this function could be changed to parse `""` as 0, and users below could be simplified to not special case the empty value.